core/src/main/java/io/fabric8/maven/core/util/MavenUtil.java

```diff
@@ -204,6 +222,78 @@ public static Properties getPropertiesFile(MavenProject project, String properti
     }
 
     /**
+     * Returns a {@code Properties} representation of the given Yaml file on the project classpath if found or an empty properties object if not
+     */
+    public static Properties getPropertiesFromYamlFile(MavenProject project, String yamlFileName) {
+        URLClassLoader compileClassLoader = getCompileClassLoader(project);
+        URL resource = compileClassLoader.findResource(yamlFileName);
+        return getPropertiesFromYamlResource(resource);
+    }
+
+    /**
+     * Returns a {@code Properties} representation of the given Yaml resource or an empty properties object if the resource is null
+     */
+    protected static Properties getPropertiesFromYamlResource(URL resource) {
+        if (resource != null) {
+            try (InputStream yamlStream = resource.openStream()) {
+                Yaml yaml = new Yaml();
+                @SuppressWarnings("unchecked")
+                SortedMap<String, Object> source = yaml.loadAs(yamlStream, SortedMap.class);
+                Properties properties = new Properties();
+                properties.putAll(getFlattenedMap(source));
+                return properties;
+            } catch (Exception e) {
+                LOG.error("Failed to load yaml properties from: " + resource + ". " + e, e);
```

Comment:
The general question is, whether we should really only log the error or wouldn't it be better for this utility class to propagate a IOException to its called. Logging in utility classes is an antipattern anyway, so I would avoid that. (I know this has been there previously already, but would like to clean it now anyway :)

So I would simply remove the catch block and let the exception propagate.